### PR TITLE
Docker! 

### PR DESCRIPTION
**Pull request by katjad**
Dockerfile, .docker-compose and .dockerignore :) I will need to update the README with instructions another time, too much for today! 

There's two more commits in the PR about adding UIkit and template updates. Have not  looked at UIkit so much yet, but looks like it might be good.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - backend
+
+  web:
+    build: .
+    restart: always
+    depends_on:
+      - db
+    ports:
+      - 8000:8000
+    environment:
```
Comment (Geekfish) [inline docker-compose.yml:23]: do you need these now that your server loads the .env-file ?

File: bookx/settings.py [hunk 2]
```diff
@@ -144,4 +144,7 @@
 # this is a bit confusing because it is not the url
 # but the name used for reverse lookup
 LOGIN_REDIRECT_URL = 'home'
-LOGOUT_REDIRECT_URL = 'home'
\ No newline at end of file
+LOGOUT_REDIRECT_URL = 'home'
+
+# custom setting by kd
+UIKIT_ROOT = os.path.join(BASE_DIR, 'static/vendor/uikit-3.0.0-rc.20/')
```
Comment (Geekfish) [inline bookx/settings.py:148]: Is there a reason for this setting?

**Review by Geekfish — APPROVED** [review]
🚢 🚢 🚢 Looks good! I hope I didn't miss anything, Github can't really deal with the number of files added for UI Kit. I'm really keen to learn more about it as well.

I have never used `.dockerignore` before, I haven't had any problems with pycache / pyc etc, I suspect because the directory is just mounted and any leftovers are generated by the container itself, not coming from another system... I'm wondering if they can cause issues!

File: docker-compose.yml (hunk 1 above)
Comment (katjad) [reply docker-compose.yml:23]: No probably not, will check and remove in that case.

File: bookx/settings.py (hunk 2 above)
Comment (katjad) [reply bookx/settings.py:148]: Mmh I think I might have put it there so I can us it in a template or something, but I can't remember what the exact reason was!

**Comment by katjad** [comment]
Thanks for merging!! I only saw this now, was at Mark's in the afternoon and did not check emails till now :) I have been working on instructions for manually setting up which has been interesting. Want to add docker instructions now. Hopefully next PR incoming soon!

**Pull request merged**